crates/services/consensus_module/poa/src/service_test/manually_produce_tests.rs

```diff
@@ -6,30 +6,39 @@ use test_case::test_case;
-#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Instant, 0)]
+#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Never, 0;
+"can manually produce blocks even when trigger is Never")]
+#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Instant, 0;
+"can manually produce blocks when trigger is Instant")]
 #[test_case(
     Tai64::now(), 3, vec![Tai64::now(), Tai64::now() + 10, Tai64::now() + 20],
     Trigger::Interval { block_time: Duration::from_secs(10) }, 0
-)]
-#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Never, 0)]
-#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Instant, 0)]
+; "can manually produce blocks with different times")]
+#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Never, 0;
+"can manually produce blocks starting in the future even when trigger is Never")]
+#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Instant, 0;
+"can manually produce blocks starting in the future when trigger is Instant")]
 #[test_case(
     Tai64::now() + 100, 3, vec![Tai64::now() + 100, Tai64::now() + 110, Tai64::now() + 120],
-    Trigger::Interval { block_time: Duration::from_secs(10) }, 0
-)]
-#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Never, 10)]
-#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Instant, 10)]
+    Trigger::Interval { block_time: Duration::from_secs(10) }, 0;
+"can manually produce blocks starting in the future with different times")]
+#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Never, 10;
+"can manually produce blocks with txs even when trigger is Never")]
+#[test_case(Tai64::now(), 10, vec![Tai64::now(); 10], Trigger::Instant, 10;
+"can manually produce blocks with txs when trigger is Instant")]
 #[test_case(
     Tai64::now(), 3, vec![Tai64::now(), Tai64::now() + 10, Tai64::now() + 20],
     Trigger::Interval { block_time: Duration::from_secs(10) }, 10
-)]
-#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Never, 10)]
-#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Instant, 10)]
+;
+"can manually produce blocks with different times with txs")]
+#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Never, 10;
+"can manually produce blocks with txs starting in the future even when trigger is Never")]
+#[test_case(Tai64::now() + 100, 10, vec![Tai64::now() + 100; 10], Trigger::Instant, 10;
+"can manually produce blocks with txs starting in the future when trigger is Instant")]
 #[test_case(
     Tai64::now() + 100, 3, vec![Tai64::now() + 100, Tai64::now() + 110, Tai64::now() + 120],
-    Trigger::Interval { block_time: Duration::from_secs(10) }, 10
-)]
+    Trigger::Interval { block_time: Duration::from_secs(10) }, 10;
+"can manually produce blocks with txs starting in the future with different times")]
 #[tokio::test]
 async fn can_manually_produce_block(
```

Comment:
I'm interested to learn why we have tests for manually producing blocks. Who is the customer for manually producing blocks?

Reply:
It is helpful for developers to test cases where you need time travel. Like you deployed a contract/called some method at height X, and at height Y, you need to test some behavior =)

Reply:
Cool. That's what I suspected.
```
let production_enabled =
        !matches!(poa_config.trigger, Trigger::Never) || config.manual_blocks_enabled;
```
It seems like the manual production logic is exposed in production code. I don't think that should be the case. I like the idea of having it for tests :100: , but it should be hidden behind Block Production interface in some `TestBlockProducer` adapter.